zigpy/zcl/clusters/greenpower.py

```diff
@@ -0,0 +1,331 @@
+"""Zigbee Green Power Domain"""
+
+from __future__ import annotations
+
+from typing import Final
+
+import zigpy.types as t
+from zigpy.types.struct import StructField
+from zigpy.zcl import Cluster, foundation
+from zigpy.zcl.foundation import (
+    BaseAttributeDefs,
+    BaseCommandDefs,
+    ZCLAttributeDef,
+    ZCLCommandDef,
+)
+import zigpy.zgp.types as zgptypes
+
+
+class CommissioningNotificationOptions(t.Struct):
+    application_id: zgptypes.ApplicationID
+    rx_after_tx: t.uint1_t
+    security_level: zgptypes.SecurityLevel
+    security_key_type: zgptypes.SecurityKeyType
+    security_failed: t.uint1_t
+    bidirectional_cap: t.uint1_t
+    proxy_info_present: t.uint1_t
+    _reserved: t.uint6_t
+
+
+# Figure 27
+class CommissioningNotificationSchema(foundation.CommandSchema):
+    options: CommissioningNotificationOptions
+    gpd_id: zgptypes.DeviceID
+    frame_counter: t.uint32_t
+    command_id: t.uint8_t
+    payload: t.LVBytes
+    gpp_short_addr: t.uint16_t = StructField(
+        requires=lambda s: s.proxy_info_present, optional=True
+    )
+    distance: t.uint8_t = StructField(
+        requires=lambda s: s.proxy_info_present, optional=True
+    )
+    mic: t.uint32_t = StructField(requires=lambda s: s.security_failed, optional=True)
+
+
+class ResponseOptions(t.Struct):
+    application_id: zgptypes.ApplicationID
+    _reserved: t.uint5_t
```

Comment:
Should these be `Struct` types, or `IntStruct` types? Also: if they're `IntStruct` types with fixed width, can the reserved fields be removed?